core/store/src/contract.rs

```diff
@@ -24,61 +24,159 @@ pub struct ContractStorage {
     /// The State this field should be removed, and the `Storage::store` function should be
     /// adjusted to write out the contract into the relevant part of the database immediately
     /// (without going through transactional storage operations and such).
-    uncommitted_deploys: Arc<Mutex<BTreeMap<CryptoHash, ContractCode>>>,
+    uncommitted_deploys: BTreeMap<CodeHash, ContractCode>,
+
+    /// Deployments moved from `uncommitted_deploys` by calling `commit`.
+    /// These are returned from `finalize`.
+    committed_deploys: BTreeMap<CodeHash, ContractCode>,
+
+    /// List of code-hashes for the contracts called.
+    ///
+    /// We do not distinguish between committed and uncommitted calls, because we need
+    /// to record all calls to validate both successful and failing function calls.
+    contract_calls: BTreeSet<CodeHash>,
+}
+
+impl ContractsTracker {
+    fn get(&self, code_hash: CodeHash) -> Option<ContractCode> {
+        self.uncommitted_deploys
+            .get(&code_hash)
+            .or_else(|| self.committed_deploys.get(&code_hash))
+            .and_then(|contract| {
```

Comment:
probably you wanted `.map` here

Reply:
yes, better to avoid explicit Some. done.